### PR TITLE
fixes(configuration): Not unique exporter for exporter type 

### LINKED ISSUE #1700: [Bug] Not unique exporter for exporter type
## Description

If the same exporter is defined in several levels (assembly and class) with different parameters, they are not merged.

## Steps to Reproduce

1. Download attachment
2. Decompress
3. Run
4. At the end of benchmark you can see in Console output like this:
   ```shell
   // * Export *
   BenchmarkDotNet.Artifacts\results\TestMultipleExportIssue.SimpleBenchmark-report.csv
   BenchmarkDotNet.Artifacts\results\TestMultipleExportIssue.SimpleBenchmark-report-github.md
   BenchmarkDotNet.Artifacts\results\TestMultipleExportIssue.SimpleBenchmark-report.html
   CustomExporter - Benchmark
   CustomExporter - Assembly
   ```

## Expected Behavior
Console output like this:
```shell
// * Export *
BenchmarkDotNet.Artifacts\results\TestMultipleExportIssue.SimpleBenchmark-report.csv
BenchmarkDotNet.Artifacts\results\TestMultipleExportIssue.SimpleBenchmark-report-github.md
BenchmarkDotNet.Artifacts\results\TestMultipleExportIssue.SimpleBenchmark-report.html
CustomExporter - Benchmark
```

Version: [0.12.1](https://github.com/dotnet/BenchmarkDotNet/releases/tag/v0.12.1)

### Attachment
[TestMultipleExportIssue.zip](https://github.com/dotnet/BenchmarkDotNet/files/6440779/TestMultipleExportIssue.zip)

### PR DESCRIPTION
**Pull request by workgroupengineering**
fixes #1700

alternative version of PR #1702

### CODE REVIEW

File: src/BenchmarkDotNet/Running/BenchmarkRunnerClean.cs [hunk 1]
```diff
@@ -237,6 +237,9 @@ private static void PrintSummary(ILogger logger, ImmutableConfig config, Summary
             // TODO: make exporter
             ConclusionHelper.Print(logger, config.GetCompositeAnalyser().Analyse(summary).Distinct().ToList());
 
+            logger.WriteLineHeader("// * Config Issues *");
```
Comment (AndreyAkinshin) [inline src/BenchmarkDotNet/Running/BenchmarkRunnerClean.cs:240]: We should omit the header if there are no any issues.

File: src/BenchmarkDotNet/Configs/DebugConfig.cs [hunk 2]
```diff
@@ -82,5 +83,7 @@ public string ArtifactsPath
         public IEnumerable<BenchmarkLogicalGroupRule> GetLogicalGroupRules() => Array.Empty<BenchmarkLogicalGroupRule>();
 
         public ConfigOptions Options => ConfigOptions.KeepBenchmarkFiles | ConfigOptions.DisableOptimizationsValidator;
+
+        public IReadOnlyList<Conclusion> ConfigAnalyse => emptyConclusion;
```
Comment (AndreyAkinshin) [inline src/BenchmarkDotNet/Configs/DebugConfig.cs:87]: `ConfigAnalyse` sounds a little bit confusing. Could we rename it to something like `ConfigConclusions` or `ConfigAnalysisConclusion`?

**Review by AndreyAkinshin — CHANGES_REQUESTED** [review]

**Review by AndreyAkinshin — APPROVED** [review]

**Comment by AndreyAkinshin** [comment]
@workgroupengineering thanks!

File: src/BenchmarkDotNet/Configs/ManualConfig.cs [hunk 3]
```diff
@@ -29,7 +29,8 @@ public class ManualConfig : IConfig
         private readonly List<Job> jobs = new List<Job>();
         private readonly HashSet<HardwareCounter> hardwareCounters = new HashSet<HardwareCounter>();
         private readonly List<IFilter> filters = new List<IFilter>();
-        private readonly List<BenchmarkLogicalGroupRule> logicalGroupRules = new List<BenchmarkLogicalGroupRule>();
+        private readonly HashSet<BenchmarkLogicalGroupRule> logicalGroupRules = new HashSet<BenchmarkLogicalGroupRule>();
+        private readonly static Conclusion[] emptyConclusion = Array.Empty<Conclusion>();
```
Comment (mawosoft) [inline src/BenchmarkDotNet/Configs/ManualConfig.cs:33]: @AndreyAkinshin @workgroupengineering
Not sure if you noticed, but this turns logicalGroupRules back into HashSet<> (changed to List<> in #1866).
(I didn't do a full review, but this one jumped at me)

Comment (AndreyAkinshin) [reply src/BenchmarkDotNet/Configs/ManualConfig.cs:33]: @mawosoft nice catch! Fixed in bd08722ee22fc25a0cf6d389125c5b6eff10cd8f

**Pull request merged**